x/staking/keeper/val_state_change.go

```diff
@@ -112,6 +116,11 @@ func (k Keeper) BlockValidatorUpdates(ctx context.Context) ([]abci.ValidatorUpda
 		)
 	}
 
+	err = k.PurgeAllMaturedConsKeyRotatedKeys(sdkCtx, sdkCtx.HeaderInfo().Time)
```

Comment:
What if the list of matured cons key is very big. Given that this is called in enblocker, should be add a limit?

Reply:
I believe we have a limit to how many rotations a validator can execute. So in theory, you can devise a formula for a cap.

Reply:
Yes we do have limit, the current limit is 1 rotation per unbonding period.